src/components/TextMiddleEllipsis.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { HTMLAttributes, MutableRefObject, useEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+interface TextMiddleEllipsisProps extends HTMLAttributes<HTMLDivElement> {
+  text: string
+  className?: string
+}
+
+const TextMiddleEllipsis = ({ text, className }: TextMiddleEllipsisProps) => {
+  const el = useRef<HTMLDivElement | null>(null)
+  const charWidth = useRef<number>()
+  const [_text, setText] = useState(text)
+
+  const handleResize = createHandleResize(el, charWidth, text, setText)
+
+  useEffect(() => {
+    handleResize()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [text])
+
+  useEffect(() => {
+    window.addEventListener('resize', handleResize)
+    return () => window.removeEventListener('resize', handleResize)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  return (
+    <div ref={el} className={className}>
+      <HiddenText>{text}</HiddenText>
+      <div>{_text}</div>
+    </div>
+  )
+}
+
+const createHandleResize =
+  (
+    el: MutableRefObject<HTMLDivElement | null>,
+    charWidth: MutableRefObject<number | undefined>,
+    text: string,
+    setText: (t: string) => void
+  ) =>
+  () => {
+    if (el?.current === null) return
+
+    if (charWidth.current === undefined) {
+      charWidth.current = el.current.scrollWidth / text.length
+    }
+
+    const visibleChars = Math.floor(el.current.clientWidth / charWidth.current)
+    const half = visibleChars / 2
+
+    setText(
+      visibleChars >= text.length
+        ? text
+        : text.slice(0, Math.floor(half)) +
+            (visibleChars == text.length ? '' : '...') +
+            text.slice(-Math.ceil(half) + 3)
+    )
```

Comment:
I'm pretty sure this would have had the same effect without having to pass the state and set state functions are arguments. But I guess this was just copied from another project so maybe we can just add a comment to simplify it later, to not keep this PR stale.

```ts
  const handleResize = useCallback(() => {
    if (el?.current === null) return

    if (charWidth.current === undefined) {
      charWidth.current = el.current.scrollWidth / text.length
    }

    const visibleChars = Math.floor(el.current.clientWidth / charWidth.current)
    const half = visibleChars / 2

    setText(
      visibleChars >= text.length
        ? text
        : text.slice(0, Math.floor(half)) +
            (visibleChars == text.length ? '' : '...') +
            text.slice(-Math.ceil(half) + 3)
    )
  }, [text])

  useEffect(() => {
    handleResize()
  }, [handleResize])
```

Reply:
I do. I am curius, however, what the future of js-sdk is. Now that we are moving parts of it to @alephium/web3, js-sdk might be a good candidate to host such components that are useful for frontend apps, since all it will contain is util functions for building frontends. What are your thoughts on this?

Reply:
I'm on my way to Neuch right now, hehe! But I'll be there next week as well.

I think I'd prefer a single repo, I don't see it as 180 turn, since js-sdk becomes more and more a "wallet/explorer"-sdk. But anyway, looking forward to discussing approaches next week!

I've approved the PR.

Reply:
I've created a repo... juste in case :) For now, it's used as a placeholder for discussions. Cheers!